contrib/apparmor/main.go

```diff
@@ -0,0 +1,76 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path"
+	"strconv"
+	"strings"
+	"text/template"
+)
+
+type profileData struct {
+	Version float64
+}
+
+func main() {
+	if len(os.Args) < 2 {
+		log.Fatal("pass a filename to save the profile in.")
+	}
+
+	// parse the arg
+	apparmorProfilePath := os.Args[1]
+
+	// get the apparmor_version version
+	cmd := exec.Command("/sbin/apparmor_parser", "--version")
+
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		log.Fatalf("getting apparmor_parser version failed: %s (%s)", err, output)
+	}
+
+	// parse the version from the output
+	// output is in the form of the following:
+	// AppArmor parser version 2.9.1
+	// Copyright (C) 1999-2008 Novell Inc.
+	// Copyright 2009-2012 Canonical Ltd.
+	lines := strings.SplitN(string(output), "\n", 2)
+	words := strings.Split(lines[0], " ")
+	version := words[len(words)-1]
+	// get major minor version only
+	if strings.Count(version, ".") > 1 {
+		version = version[:strings.LastIndex(version, ".")]
+	}
+
+	v, err := strconv.ParseFloat(version, 64)
```

Comment:
facepalm was trying to avoid using a version package like ours

On Fri, Oct 16, 2015 at 4:57 PM, Jessie Frazelle <jess@docker.com> wrote:

> oh shoot
>
> On Fri, Oct 16, 2015 at 4:54 PM, Tianon Gravi <notifications@github.com>
> wrote:
>
>> In contrib/apparmor/main.go
>> <https://github.com/docker/docker/pull/17002#discussion_r42301473>:
>>
>> > +	}
>> > +
>> > +	// parse the version from the output
>> > +	// output is in the form of the following:
>> > +	// AppArmor parser version 2.9.1
>> > +	// Copyright (C) 1999-2008 Novell Inc.
>> > +	// Copyright 2009-2012 Canonical Ltd.
>> > +	lines := strings.SplitN(string(output), "\n", 2)
>> > +	words := strings.Split(lines[0], " ")
>> > +	version := words[len(words)-1]
>> > +	// get major minor version only
>> > +	if strings.Count(version, ".") > 1 {
>> > +		version = version[:strings.LastIndex(version, ".")]
>> > +	}
>> > +
>> > +	v, err := strconv.ParseFloat(version, 64)
>>
>> Won't this mean that 3.9 > 3.10 if we just do the naïve float conversion
>> here like this?
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/docker/docker/pull/17002/files#r42301473>.
>>
>
>